### ISSUE TITLE
useRouteQuery should support array

### ISSUE THREAD

**Issue by ottopaulsen** [vueuse/vueuse#3878]
### Clear and concise description of the problem

Some times a query param can have multiple values: ?state=NEW&state=OLD
In this case you may want to always get an array if you read the query param, either an empty array or an array with one or more values. Two in the example.
However, what you get is a string if there is only one value, and an array only if there are multiple values. If you use useRouteQuery with `{transform: Array}` you get an array if there is 0 or 1 values, but if there are multiple values you get an array with an array.

### Suggested solution

Fix so if `{transform: Array}` is used, it always returns a flattened array.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by OrbisK** [comment]
I agree with @Alexance . 

In addition, there are also several conventions on how to display arrays in the query. I would close the issue for now. If there is still a need, please feel free to open it again or start a discussion.

**Comment by rasyidly** [comment]
Hi, I would like to suggest adding a new utility function called `useRouteQueryArray` to the @vueuse/router package. This function would extend the existing `useRouteQuery` function by providing built-in support for handling query parameters as arrays.

The `useRouteQueryArray` function would accept the same parameters as `useRouteQuery`, but it would include an additional transform option that automatically converts single query parameter values into arrays. This would simplify the process of working with query parameters that can have multiple values, such as filters or selections in a user interface.

Here is a basic example of how the `useRouteQueryArray` function could be implemented:

```typescript
import { useRouteQuery } from '@vueuse/router'

export function useRouteQueryArray(...props: Parameters<typeof useRouteQuery<string[]>>): ReturnType<typeof useRouteQuery<string[]>> {
  return useRouteQuery<string[]>(props[0], props[1], {
    transform: v => Array.isArray(v) ? v : (v ? [v] : []),
    ...props[2]
  })
}
```

With this utility, developers can easily manage query parameters as arrays without having to manually handle the conversion logic each time. This would enhance the usability of the @vueuse/router package and streamline the development process for applications that rely heavily on query parameters.

Thank you for considering this suggestion. I believe it would be a valuable addition to the @vueuse/router package and benefit many developers in the community.

**Comment by ottopaulsen** [comment]
Good idea, @rasyidly , but this is a closed issue, so you should probably post this as a new issue.
